src/coreclr/jit/instrsarm64_sve.h

```diff
@@ -0,0 +1,2873 @@
+    // ADR     <Zd>.<T>, [<Zn>.<T>, <Zm>.<T>{, <mod><amount>}]                           SVE_BH_3A           000001001x1mmmmm 1010hhnnnnnddddd     04A0 A000   
+    // ADR     <Zd>.D, [<Zn>.D, <Zm>.D, SXTW{<amount>}]                                  SVE_BH_3B           00000100001mmmmm 1010hhnnnnnddddd     0420 A000   
+    // ADR     <Zd>.D, [<Zn>.D, <Zm>.D, UXTW{<amount>}]                                  SVE_BH_3B_A         00000100011mmmmm 1010hhnnnnnddddd     0460 A000   
+
+
+//    enum               name                     info                                              SVE_BW_2A        SVE_CB_2A        SVE_EB_1A        
+INST3(dup,               "dup",                   0,                       IF_SVE_3C,               0x05202000,      0x05203800,      0x2538C000       )
+    // DUP     <Zd>.<T>, <Zn>.<T>[<imm>]                                                 SVE_BW_2A           00000101ii1xxxxx 001000nnnnnddddd     0520 2000   
+    // DUP     <Zd>.<T>, <R><n|SP>                                                       SVE_CB_2A           00000101xx100000 001110nnnnnddddd     0520 3800   
+    // DUP     <Zd>.<T>, #<imm>{, <shift>}                                               SVE_EB_1A           00100101xx111000 11hiiiiiiiiddddd     2538 C000   
+
+
+//    enum               name                     info                                              SVE_BR_3A        SVE_BR_3B        SVE_CI_3A        
+INST3(trn1,              "trn1",                  0,                       IF_SVE_3D,               0x05207000,      0x05A01800,      0x05205000       )
+    // TRN1    <Zd>.<T>, <Zn>.<T>, <Zm>.<T>                                              SVE_BR_3A           00000101xx1mmmmm 011100nnnnnddddd     0520 7000   
+    // TRN1    <Zd>.Q, <Zn>.Q, <Zm>.Q                                                    SVE_BR_3B           00000101101mmmmm 000110nnnnnddddd     05A0 1800   
+    // TRN1    <Pd>.<T>, <Pn>.<T>, <Pm>.<T>                                              SVE_CI_3A           00000101xx10MMMM 0101000NNNN0DDDD     0520 5000   
+
+INST3(trn2,              "trn2",                  0,                       IF_SVE_3D,               0x05207400,      0x05A01C00,      0x05205400       )
+    // TRN2    <Zd>.<T>, <Zn>.<T>, <Zm>.<T>                                              SVE_BR_3A           00000101xx1mmmmm 011101nnnnnddddd     0520 7400   
+    // TRN2    <Zd>.Q, <Zn>.Q, <Zm>.Q                                                    SVE_BR_3B           00000101101mmmmm 000111nnnnnddddd     05A0 1C00   
+    // TRN2    <Pd>.<T>, <Pn>.<T>, <Pm>.<T>                                              SVE_CI_3A           00000101xx10MMMM 0101010NNNN0DDDD     0520 5400   
+
+INST3(uzp1,              "uzp1",                  0,                       IF_SVE_3D,               0x05206800,      0x05A00800,      0x05204800       )
```

Comment:
Comparing this against the UZP1 I manually added https://github.com/dotnet/runtime/pull/93067

I'm a little confused by `00000101xx1mmmmm` I can't see what bit pattern that refers to.

Reply:
The changes [here](https://github.com/dotnet/runtime/pull/94285/files#diff-5e27623edecb939f3991300912bd8e69ca0746ad1cf4fc0dc3bbb5c761f2e029R623-R625) matches exactly to what you had in your PR [here](https://github.com/dotnet/runtime/pull/93067/files#diff-534bddc341b6c03fb9519adcc7a1d25c6d4fc2f66605cc1bae741f27a34e0212R1685-R1686).

The extra encoding `00000101101mmmmm 000010nnnnnddddd     05A0 0800 ` present is for `Even(quadwords)`:

<img width="634" alt="image" src="https://github.com/dotnet/runtime/assets/12488060/0a1691a7-39e8-426e-acdd-b725004a8c0c">

Reply:
Let me know if that answers your question.

Reply:
Got it, missed that one.